Add pristine options to "west build" and the config (#15461)
zephyrproject-rtos/zephyr: doc/guides/west/config.rst

       environment overrides the value of the ``zephyr.base`` configuration
       option. If set to ``"configfile"``, the configuration option wins
       instead.

Zephyr Extension Commands Configuration Options
-----------------------------------------------

The following table documents configuration options supported by zephyr's

> extension commands (found in :file:`scripts/west_commands`)

Reviewer: ```suggestion
extension commands (found in :file:`scripts/west_commands`).
```

.. NOTE: docs authors: keep this table sorted by section, then option.

.. list-table::
   :widths: 10 30
   :header-rows: 1

   * - Option
     - Description
   * - ``build.auto-pristine``
     - Boolean. If ``true`` (the default is ``false``), then ``west build`` will

>        automatically run the ``pristine`` target if a build system is present

Reviewer: ```suggestion
       automatically run the ``pristine`` target before building, if a build system is present
```

>        and the build will fail otherwise (eg. the user has specified a different

Reviewer: ```suggestion
       and the build will fail otherwise (e.g. the user has specified a different
```

>        board or application)

Reviewer: ```suggestion
       board or application from the one previously used to make the build directory.)
```

       environment overrides the value of the ``zephyr.base`` configuration
       option. If set to ``"configfile"``, the configuration option wins
       instead.

Zephyr Extension Commands Configuration Options
-----------------------------------------------

> The following table documents configuration options supported by zephyr's

Reviewer 2: ``Zephyr`` (upper case)

Author: Actually I intentionally used lower-case because I am referring to the repository, since extension commands are per-repository.
